depthmapXcli/segmentparser.cpp

```diff
@@ -0,0 +1,151 @@
+            {
+                throw CommandLineException("-st can only be used once, modes are mutually exclusive");
+            }
+            ENFORCE_ARGUMENT("-st", i)
+            if ( std::strcmp(argv[i], "tulip") == 0 )
+            {
+                _analysisType = AnalysisType::ANGULAR_TULIP;
+            }
+            else if ( std::strcmp(argv[i], "angular") == 0 )
+            {
+                _analysisType = AnalysisType::ANGULAR_FULL;
+            }
+            else if ( std::strcmp(argv[i], "topological") == 0 )
+            {
+                _analysisType = AnalysisType::TOPOLOGICAL;
+            }
+            else if ( std::strcmp(argv[i], "metric") == 0 )
+            {
+                _analysisType = AnalysisType::METRIC;
+            }
+        }
+        if (std::strcmp(argv[i], "-srt") == 0)
+        {
+            if (_radiusType != RadiusType::NONE)
+            {
+                throw CommandLineException("-srt can only be used once, modes are mutually exclusive");
+            }
+            ENFORCE_ARGUMENT("-srt", i)
+            if ( std::strcmp(argv[i], "steps") == 0 )
+            {
+                _radiusType = RadiusType::SEGMENT_STEPS;
+            }
+            else if ( std::strcmp(argv[i], "angular") == 0 )
+            {
+                _radiusType = RadiusType::ANGULAR;
+            }
+            else if ( std::strcmp(argv[i], "metric") == 0 )
+            {
+                _radiusType = RadiusType::METRIC;
+            }
+        }
+        else if(std::strcmp(argv[i], "-sic") == 0)
+        {
+            _includeChoice = true;
+        }
+        else if (std::strcmp(argv[i], "-sr") == 0)
+        {
+            ENFORCE_ARGUMENT("-sr", i)
+            _radii = depthmapX::parseRadiusList(argv[i]);
+        }
+        else if (std::strcmp(argv[i], "-stb") == 0)
+        {
+            ENFORCE_ARGUMENT("-stb", i)
+
+            if (!has_only_digits(argv[i]))
+            {
+                throw CommandLineException(std::string("-stb must be a number between 4 and 1024, got ") + argv[i]);
+            }
+            _tulipBins = std::atoi(argv[i]);
+            if (_tulipBins < 4 || _tulipBins > 1024)
+            {
+                throw CommandLineException(std::string("-stb must be a number between 4 and 1024, got ") + argv[i]);
+            }
+        }
+    }
+
+    if (getAnalysisType() == AnalysisType::NONE)
+    {
+        throw CommandLineException("No analysis type given");
+    }
+
+    if (getRadii().empty())
+    {
+        throw CommandLineException("At least one radius must be provided");
+    }
+
+    if (getAnalysisType() == AnalysisType::ANGULAR_TULIP && getRadiusType() == RadiusType::NONE)
+    {
+        throw CommandLineException("Radius type is required for tulip analysis");
+    }
+
+    if (getAnalysisType() == AnalysisType::ANGULAR_TULIP && getTulipBins() == 0)
+    {
+        throw CommandLineException("Tulip bins are required for tulip analysis");
+    }
+}
```

Comment:
You're missing to check the case where tulipBins have been set but the type is `ANGULAR_FULL` - this will silently run a tulip analysis while claiming to run a full one!

Reply:
Addressed in commit 10a6a18 by splitting the two functions in the MetaGraph. Should the code throw if the user selects full-angular AND sets tulip bins?

Reply:
Yes. If the user sets an options and expects this to change something, but it doesn't make sense, we should let them know, not just plough on.